{
    "GSD": {
        "alias": "CVE-2017-1000246",
        "description": "Python package pysaml2 version 4.4.0 and earlier reuses the initialization vector across encryptions in the IDP server, resulting in weak encryption of data.",
        "id": "GSD-2017-1000246",
        "references": [
            "https://www.suse.com/security/cve/CVE-2017-1000246.html"
        ]
    },
    "namespaces": {
        "cve.org": {
            "CVE_data_meta": {
                "ASSIGNER": "cve@mitre.org",
                "DATE_ASSIGNED": "2017-08-22T17:29:33.475589",
                "ID": "CVE-2017-1000246",
                "REQUESTER": "predrag.gruevski@gmail.com",
                "STATE": "PUBLIC"
            },
            "affects": {
                "vendor": {
                    "vendor_data": [
                        {
                            "product": {
                                "product_data": [
                                    {
                                        "product_name": "n/a",
                                        "version": {
                                            "version_data": [
                                                {
                                                    "version_value": "n/a"
                                                }
                                            ]
                                        }
                                    }
                                ]
                            },
                            "vendor_name": "n/a"
                        }
                    ]
                }
            },
            "data_format": "MITRE",
            "data_type": "CVE",
            "data_version": "4.0",
            "description": {
                "description_data": [
                    {
                        "lang": "eng",
                        "value": "Python package pysaml2 version 4.4.0 and earlier reuses the initialization vector across encryptions in the IDP server, resulting in weak encryption of data."
                    }
                ]
            },
            "problemtype": {
                "problemtype_data": [
                    {
                        "description": [
                            {
                                "lang": "eng",
                                "value": "n/a"
                            }
                        ]
                    }
                ]
            },
            "references": {
                "reference_data": [
                    {
                        "name": "https://github.com/rohe/pysaml2/issues/417",
                        "refsource": "MISC",
                        "url": "https://github.com/rohe/pysaml2/issues/417"
                    }
                ]
            }
        },
        "gitlab.com": {
            "advisories": [
                {
                    "affected_range": "<4.6.0",
                    "affected_versions": "All versions before 4.6.0",
                    "cvss_v2": "AV:N/AC:L/Au:N/C:P/I:N/A:N",
                    "cvss_v3": "CVSS:3.0/AV:N/AC:L/PR:N/UI:N/S:U/C:L/I:N/A:N",
                    "cwe_ids": [
                        "CWE-1035",
                        "CWE-330",
                        "CWE-937"
                    ],
                    "date": "2019-10-03",
                    "description": "Python package pysaml2 reuses the initialization vector across encryptions in the IDP server, resulting in weak encryption of data.",
                    "fixed_versions": [
                        "4.6.0"
                    ],
                    "identifier": "CVE-2017-1000246",
                    "identifiers": [
                        "CVE-2017-1000246"
                    ],
                    "not_impacted": "All versions starting from 4.6.0",
                    "package_slug": "pypi/pysaml2",
                    "pubdate": "2017-11-17",
                    "solution": "Upgrade to version 4.6.0 or above.",
                    "title": "Use of Insufficiently Random Values",
                    "urls": [
                        "https://nvd.nist.gov/vuln/detail/CVE-2017-1000246",
                        "https://github.com/rohe/pysaml2/issues/417"
                    ],
                    "uuid": "40eff80e-5f42-4738-b7b2-c7b1a52f76fd"
                }
            ]
        },
        "nvd.nist.gov": {
            "configurations": {
                "CVE_data_version": "4.0",
                "nodes": [
                    {
                        "children": [],
                        "cpe_match": [
                            {
                                "cpe23Uri": "cpe:2.3:a:pysaml2_project:pysaml2:*:*:*:*:*:*:*:*",
                                "cpe_name": [],
                                "versionEndExcluding": "4.6.0",
                                "vulnerable": true
                            }
                        ],
                        "operator": "OR"
                    }
                ]
            },
            "cve": {
                "CVE_data_meta": {
                    "ASSIGNER": "cve@mitre.org",
                    "ID": "CVE-2017-1000246"
                },
                "data_format": "MITRE",
                "data_type": "CVE",
                "data_version": "4.0",
                "description": {
                    "description_data": [
                        {
                            "lang": "en",
                            "value": "Python package pysaml2 version 4.4.0 and earlier reuses the initialization vector across encryptions in the IDP server, resulting in weak encryption of data."
                        }
                    ]
                },
                "problemtype": {
                    "problemtype_data": [
                        {
                            "description": [
                                {
                                    "lang": "en",
                                    "value": "CWE-330"
                                }
                            ]
                        }
                    ]
                },
                "references": {
                    "reference_data": [
                        {
                            "name": "https://github.com/rohe/pysaml2/issues/417",
                            "refsource": "MISC",
                            "tags": [
                                "Issue Tracking",
                                "Patch",
                                "Third Party Advisory"
                            ],
                            "url": "https://github.com/rohe/pysaml2/issues/417"
                        }
                    ]
                }
            },
            "impact": {
                "baseMetricV2": {
                    "acInsufInfo": true,
                    "cvssV2": {
                        "accessComplexity": "LOW",
                        "accessVector": "NETWORK",
                        "authentication": "NONE",
                        "availabilityImpact": "NONE",
                        "baseScore": 5.0,
                        "confidentialityImpact": "PARTIAL",
                        "integrityImpact": "NONE",
                        "vectorString": "AV:N/AC:L/Au:N/C:P/I:N/A:N",
                        "version": "2.0"
                    },
                    "exploitabilityScore": 10.0,
                    "impactScore": 2.9,
                    "obtainAllPrivilege": false,
                    "obtainOtherPrivilege": false,
                    "obtainUserPrivilege": false,
                    "severity": "MEDIUM",
                    "userInteractionRequired": false
                },
                "baseMetricV3": {
                    "cvssV3": {
                        "attackComplexity": "LOW",
                        "attackVector": "NETWORK",
                        "availabilityImpact": "NONE",
                        "baseScore": 5.3,
                        "baseSeverity": "MEDIUM",
                        "confidentialityImpact": "LOW",
                        "integrityImpact": "NONE",
                        "privilegesRequired": "NONE",
                        "scope": "UNCHANGED",
                        "userInteraction": "NONE",
                        "vectorString": "CVSS:3.0/AV:N/AC:L/PR:N/UI:N/S:U/C:L/I:N/A:N",
                        "version": "3.0"
                    },
                    "exploitabilityScore": 3.9,
                    "impactScore": 1.4
                }
            },
            "lastModifiedDate": "2019-10-03T00:03Z",
            "publishedDate": "2017-11-17T04:29Z"
        }
    }
}